crates/bindings-csharp/Codegen/Module.cs

```diff
@@ -117,21 +175,122 @@ public TableDeclaration(GeneratorAttributeSyntaxContext context)
             throw new InvalidOperationException("Tagged enums cannot be tables.");
         }
 
-        var attr = context.Attributes.Single().ParseAs<TableAttribute>();
+        Visibility = context.TargetSymbol.DeclaredAccessibility;
+        switch (Visibility)
+        {
+            case Accessibility.ProtectedAndInternal:
+            case Accessibility.NotApplicable:
```

Comment:
This will include private types if a table class is nested inside another class, because for nested types the default visibility is private.

Is that intentional?

Reply:
Do you mean
`private class Outer { public class Inner {} }`
or
`public class Outer { private class Inner {} }`
?

Only the former should allow [Table] attributes, and being part of a private Outer should make codegen emit an error. Yeah I'll have to add something here.